modules/internal/ChapelRange.chpl

```diff
@@ -1296,6 +1296,50 @@ module ChapelRange {
   //# Parallel Iterators
   //#
 
+  iter range.these(param tag: iterKind) where tag == iterKind.standalone &&
+                                              !localeModelHasSublocales
+  {
+    if ! isBoundedRange(this) {
+      compilerError("parallel iteration is not supported over unbounded ranges");
+    }
+    if this.isAmbiguous() {
+      __primitive("chpl_error", "these -- Attempt to iterate over a range with ambiguous alignment.");
+    }
+    if debugChapelRange {
+      writeln("*** In range standalone iterator:");
+    }
+
+    const v = this.length;
+    const numChunks = if __primitive("task_get_serial") then
+                      1 else _computeNumChunks(v);
+
+    if debugChapelRange {
+      writeln("*** RI: length=", v, " numChunks=", numChunks);
+    }
+
+    if v == 0 {
+      return;
+    }
+
+    if numChunks == 1 {
+      for i in this {
+        yield i;
+      }
+    } else {
+      coforall chunk in 0..#numChunks {
```

Comment:
You *could* change this to `coforall chunk in 0..numChunks-1` in order to get the anonymous range optimization to fire. I don't actually think that's a good idea though, I think I need to update the anonymous range optimization to handle cases like this. Unfortunately I think that's a bit big for this release since it would require making different builders for low and high bounded ranges. :(